tests/test_samples.py

```diff
@@ -366,42 +361,35 @@ def test_plot_1d_colours():
     mn = read_chains("./tests/example_data/mn")
     mn.drop(columns='x2', inplace=True, level=0)
 
-    kinds = ['kde', 'hist']
-    if 'fastkde' in sys.modules:
-        kinds += ['fastkde']
-
-    for kind in kinds:
-        fig = plt.figure()
-        fig, axes = make_1d_axes(['x0', 'x1', 'x2', 'x3', 'x4'], fig=fig)
-        gd.plot_1d(axes, kind=kind + '_1d', label="gd")
-        pc.plot_1d(axes, kind=kind + '_1d', label="pc")
-        mn.plot_1d(axes, kind=kind + '_1d', label="mn")
-        gd_colors = []
-        pc_colors = []
-        mn_colors = []
-        for x, ax in axes.items():
-            handles, labels = ax.get_legend_handles_labels()
-            for handle, label in zip(handles, labels):
-                if isinstance(handle, Rectangle):
-                    color = to_hex(handle.get_facecolor())
-                else:
-                    color = handle.get_color()
-
-                if label == 'gd':
-                    gd_colors.append(color)
-                elif label == 'pc':
-                    pc_colors.append(color)
-                elif label == 'mn':
-                    mn_colors.append(color)
+    fig = plt.figure()
+    fig, axes = make_1d_axes(['x0', 'x1', 'x2', 'x3', 'x4'], fig=fig)
+    gd.plot_1d(axes, kind=kind + '_1d', label="gd")
+    pc.plot_1d(axes, kind=kind + '_1d', label="pc")
+    mn.plot_1d(axes, kind=kind + '_1d', label="mn")
+    gd_colors = []
+    pc_colors = []
+    mn_colors = []
+    for x, ax in axes.items():
+        handles, labels = ax.get_legend_handles_labels()
+        for handle, label in zip(handles, labels):
+            if isinstance(handle, Rectangle):
+                color = to_hex(handle.get_facecolor())
+            else:
+                color = handle.get_color()
+
+            if label == 'gd':
+                gd_colors.append(color)
+            elif label == 'pc':
+                pc_colors.append(color)
+            elif label == 'mn':
+                mn_colors.append(color)
 
-        assert len(set(gd_colors)) == 1
-        assert len(set(mn_colors)) == 1
-        assert len(set(pc_colors)) == 1
+    assert len(set(gd_colors)) == 1
+    assert len(set(mn_colors)) == 1
+    assert len(set(pc_colors)) == 1
 
 
-@pytest.mark.xfail('astropy' not in sys.modules,
-                   raises=ImportError,
-                   reason="requires astropy package")
+@astropy_mark_xfail
```

Comment:
Should this test be simply skipped rather than xfailed?

Reply:
If we don't run it, then we miss 100% coverage -- one could write specific tests which check for failure, but I think this is neater.

Reply:
Do we really miss 100% coverage? Normally our CI should be set up such that `extras: true` runs all tests without any skipping which gives us the 100% coverage. `extras: false` tests the minimal anesthetic version and is allowed to have less than 100% coverage. 

That is why the coverage tests often initially show up as failed, because the faster `extras: false` CI jobs finish first with insufficient coverage and we have to wait for the first `extras: true` to run through to get 100% coverage.

Reply:
I've changed xfails to skips so you can see the problem at [this permalink](https://app.codecov.io/gh/handley-lab/anesthetic/commit/9e159dfcdcd1cf7d37c1e87c5d9d003f31e8d1c4/blob/anesthetic/convert.py)

Reply:
Ah, ok, thanks. It is the try-except blocks in those tests, where we provide informative error messages if a package is not installed, e.g.:
```python
    try:
        import getdist
    except ModuleNotFoundError:
        raise ImportError("You need to install getdist to use to_getdist")
```
So in this getdist example the xfail with condition runs the actual test if getdist is installed (where codecov misses the except block) and checks for this ImportError if it is not (where codecov hits the except block).